src/topology/config/watcher.rs

```diff
@@ -0,0 +1,113 @@
+use super::Config;
+use futures::{Async, Future, Stream};
+use notify::{watcher, DebouncedEvent, RecursiveMode, Watcher};
+use std::{
+    path::Path,
+    sync::mpsc::{channel, TryRecvError},
+    time::{Duration, Instant},
+};
+use tokio::timer::Delay;
+
+// Per notify own documentation, it's advised to have delay of more than 30 sec,
+// so to avoid receiving repetitions of previous events on macOS.
+// Larger delays hurt responsivity, but that's fine as this is primarily designed
+// for cloud usage which isn't exactly super responsive.
+const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(31);
+
+/// Returns when config should be reloaded.
+/// Never errors, and never ends.
+pub fn config_watcher(
+    config: &Config,
+    config_path: &Path,
+) -> crate::Result<impl Stream<Item = (), Error = ()>> {
+    config_watcher_with(config, config_path, CONFIG_WATCH_DELAY)
+}
+
+fn config_watcher_with(
+    config: &Config,
+    config_path: &Path,
+    delay: Duration,
+) -> crate::Result<impl Stream<Item = (), Error = ()>> {
+    let recv_delay = delay.checked_div(2).unwrap();
```

Comment:
if we are unwrapping here does it make sense to use the checked fn?

Reply:
No need, since it's only None if divider is 0, and here it's obviously 2. 

The better question is, why did I use checked_div, ordinary / is enough.
Probably because of fever that I didn't notice at that time.

Reply:
:) I usually find I add things when I was thinking about something different then forget to remove it all the time.